### ISSUE TITLE
Expose matched route in Context (RFE)

### ISSUE THREAD

**Issue by zeisss** [kataras/iris#509]
- Version : **V4-LTS**

- [x] I have read the [book](https://docs.iris-go.com), [examples](https://github.com/iris-contrib/examples), [contributing file](https://github.com/kataras/iris/blob/master/.github/CONTRIBUTING.md) and I'm sure that this issue is not posted & answered before.

--------------

Hi,

we want to build a prometheus route in our iris-based microservice. For this we wrote a middleware which increments a prometheus counter based on the `ctx.RequestCtx.RequestURI()`. But, for parameterized routes, this includes the IDs (e.g. `/v1/tasks/12345`). 

Registering a route returns a `RouteNameFunc`, which we would love to use for the metrics, but there seems to be no way to match an URL to a route, only the other way (build an URL from a route name).

TL;DR: Expose the route name in the `Context`, for debugging & operational reasons.

**Comment by PapaStifflera** [comment]
+1

**Comment by kataras** [comment]
This is not possible by design, you have to implement this by your own and I can't accept feature requests from non- iris members, sorry
